### PR TITLE
Use executors in AccelerateBenchmark

### PR DESCRIPTION
**Pull request by satyaog**

### CODE REVIEW

File: benchmarks/accelerate_opt/benchfile.py [hunk 1]
```diff
@@ -10,79 +10,62 @@ def make_env(self):
...
-        futs.append(asyncio.create_task(self.execute(
-            *self.accelerate_command(rank=0),
-            setsid=True,
-            use_stdout=True,
-        )))
+        return SequenceExecutor(local_prepare, ListExecutor(*remote_prepare))
+
+    def build_run_plan(self):
         # XXX: this doesn't participate in the process timeout
-        for i, worker in enumerate(self.config.get('worker_addrs', [])):
```
Comment (satyaog) [inline benchmarks/accelerate_opt/benchfile.py:71]: Where is `worker_addrs` set? I would like to set it to `system.nodes[].ip`

**Comment by satyaog** [comment]
I'm puzzled to see the tests passing. This should not be the case with the config not updated right? I suppose nothing gets done w/r to `accelerate_opt`?

**Comment by satyaog** [comment]
Ah no, it fails actually in `accelerate_opt`. I will fix the code issues but I will need some guidance to set the system config right

**Comment by Delaunay** [comment]
That is the extracted command from multinode

```
opt-6_7b-multinode
accelerate launch --mixed_precision=fp16 --dynamo_backend=no --machine_rank=0 --num_machines=2 --use_deepspeed --deepspeed_multinode_launcher=standard --zero_stage=2 --gradient_accumulation_steps=1 --num_cpu_threads_per_process=8 --main_process_ip=manager-node --main_process_port=10000 --num_processes=0 /home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt/main.py
ssh -l username -i /milabench/id_milabench -o CheckHostIP=no -o StrictHostKeyChecking=no worker-node1 docker run -i --rm --network host --privileged --gpus all --env MILABENCH_CONFIG='{"system": {"arch": "cpu", "sshkey": null, "nodes": [{"name": "local", "ip": "127.0.0.1", "port": null, "user": "newton", "main": true}], "main_node": {"name": "local", "ip": "127.0.0.1", "port": null, "user": "newton", "main": true}}, "dirs": {"base": "/opt/milabench", "venv": "/opt/milabench/venv/torch", "data": "/opt/milabench/data", "runs": "/opt/milabench/runs", "extra": "/opt/milabench/extra/opt", "cache": "/opt/milabench/cache"}, "group": "opt", "install_group": "torch", "install_variant": "cpu", "run_name": "dev", "enabled": true, "capabilities": {"nodes": 2}, "max_duration": 600, "voir": {"options": {"stop": 60, "interval": "1s"}}, "validation": {"usage": {"gpu_load_threshold": 0.5, "gpu_mem_threshold": 0.5}}, "config_base": "/home/newton/work/milabench_dev/milabench/config", "config_file": "/home/newton/work/milabench_dev/milabench/config/standard.yaml", "tags": ["huggingface", "language-modeling", "llm", "multigpu", "multinode", "nlp", "transformer"], "definition": "/home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt", "plan": {"method": "njobs", "n": 1}, "manager_addr": "manager-node", "manager_port": 10000, "cpus_per_gpu": 8, "gradient_accumulation_steps": 1, "max_train_steps": 100, "dataset_name": "wikitext", "dataset_config_name": "wikitext-103-v1", "validation_split_percentage": 5, "use_deepspeed": true, "num_machines": 2, "model_name": "facebook/opt-6.7b", "per_gpu_batch_size": 1, "weight": 10.0, "requires_capabilities": ["nodes >= 2"], "docker_image": "ghcr.io/mila-iqia/milabench:cuda-nightly", "worker_user": "username", "worker_addrs": ["worker-node1", "worker-node2"], "name": "opt-6_7b-multinode", "tag": ["opt-6_7b-multinode"]}' --env OMP_NUM_THREADS='8' ghcr.io/mila-iqia/milabench:cuda-nightly /home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt/activator /opt/milabench/venv/torch accelerate launch --mixed_precision=fp16 --dynamo_backend=no --machine_rank=1 --num_machines=2 --use_deepspeed --deepspeed_multinode_launcher=standard --zero_stage=2 --gradient_accumulation_steps=1 --num_cpu_threads_per_process=8 --main_process_ip=manager-node --main_process_port=10000 --num_processes=0 /home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt/main.py
ssh -l username -i /milabench/id_milabench -o CheckHostIP=no -o StrictHostKeyChecking=no worker-node2 docker run -i --rm --network host --privileged --gpus all --env MILABENCH_CONFIG='{"system": {"arch": "cpu", "sshkey": null, "nodes": [{"name": "local", "ip": "127.0.0.1", "port": null, "user": "newton", "main": true}], "main_node": {"name": "local", "ip": "127.0.0.1", "port": null, "user": "newton", "main": true}}, "dirs": {"base": "/opt/milabench", "venv": "/opt/milabench/venv/torch", "data": "/opt/milabench/data", "runs": "/opt/milabench/runs", "extra": "/opt/milabench/extra/opt", "cache": "/opt/milabench/cache"}, "group": "opt", "install_group": "torch", "install_variant": "cpu", "run_name": "dev", "enabled": true, "capabilities": {"nodes": 2}, "max_duration": 600, "voir": {"options": {"stop": 60, "interval": "1s"}}, "validation": {"usage": {"gpu_load_threshold": 0.5, "gpu_mem_threshold": 0.5}}, "config_base": "/home/newton/work/milabench_dev/milabench/config", "config_file": "/home/newton/work/milabench_dev/milabench/config/standard.yaml", "tags": ["huggingface", "language-modeling", "llm", "multigpu", "multinode", "nlp", "transformer"], "definition": "/home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt", "plan": {"method": "njobs", "n": 1}, "manager_addr": "manager-node", "manager_port": 10000, "cpus_per_gpu": 8, "gradient_accumulation_steps": 1, "max_train_steps": 100, "dataset_name": "wikitext", "dataset_config_name": "wikitext-103-v1", "validation_split_percentage": 5, "use_deepspeed": true, "num_machines": 2, "model_name": "facebook/opt-6.7b", "per_gpu_batch_size": 1, "weight": 10.0, "requires_capabilities": ["nodes >= 2"], "docker_image": "ghcr.io/mila-iqia/milabench:cuda-nightly", "worker_user": "username", "worker_addrs": ["worker-node1", "worker-node2"], "name": "opt-6_7b-multinode", "tag": ["opt-6_7b-multinode"]}' --env OMP_NUM_THREADS='8' ghcr.io/mila-iqia/milabench:cuda-nightly /home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt/activator /opt/milabench/venv/torch accelerate launch --mixed_precision=fp16 --dynamo_backend=no --machine_rank=2 --num_machines=2 --use_deepspeed --deepspeed_multinode_launcher=standard --zero_stage=2 --gradient_accumulation_steps=1 --num_cpu_threads_per_process=8 --main_process_ip=manager-node --main_process_port=10000 --num_processes=0 /home/newton/work/milabench_dev/milabench/benchmarks/accelerate_opt/main.py
```

**Comment by satyaog** [comment]
Should be good I think? Not exactly sure why https://github.com/mila-iqia/milabench/actions/runs/5348010523/jobs/9697318334?pr=149 fails without reporting any failed bench so I'm rerunning the job:

```
=================
Benchmark results
=================
                         fail n       perf   sem%   std% peak_memory         score weight
bert-fp16                   0 2       2.16   0.2%   1.5%        7579      4.312639   0.00
bert-fp32                   0 2       2.82   0.4%   3.4%        9215      5.653523   0.00
bert-tf32                   0 2       2.82   0.4%   3.4%        9215      5.657049   0.00
bert-tf32-fp16              0 2       2.16   0.2%   1.7%        7579      4.318232   3.00
convnext_large-fp16         0 2       1.40   0.6%   4.3%        5228      2.802366   0.00
convnext_large-fp32         0 2       2.53   0.2%   1.5%        7400      5.050301   0.00
convnext_large-tf32         0 2       2.52   0.2%   1.6%        7400      5.046541   0.00
convnext_large-tf32-fp16    0 2       1.41   0.4%   3.2%        5228      2.819027   3.00
davit_large                 0 2       2.17   0.1%   0.7%       11402      4.341862   1.00
davit_large-multi           0 1       4.19   0.3%   0.8%       11972      4.187669   5.00
dlrm                        0 1   30271.44   0.9%   2.8%        1585  30271.437674   1.00
focalnet                    0 2       4.29   2.0%   8.9%        4698      8.555237   2.00
opt-6_7b                    0 1       2.52   0.1%   0.4%        3991      2.521186   5.00
reformer                    0 2       1.57   0.1%   0.4%        6471      3.131379   1.00
regnet_y_128gf              0 2       0.92   0.6%   2.7%       11705      1.841640   2.00
resnet152                   0 2      21.26   0.7%   3.1%        6062     42.515870   1.00
resnet152-multi             0 1      30.42   0.9%   2.8%        5059     30.418589   5.00
resnet50                    0 2      48.51   1.0%   4.6%        3869     97.346180   1.00
rwkv                        0 2      18.46   1.2%   5.2%        4524     36.606284   1.00
stargan                     0 2       1.37  12.4%  55.6%       11455      2.793113   1.00
super-slomo                 0 2       1.46   2.0%   9.0%        8848      2.968190   1.00
t5                          0 2       1.25   0.1%   0.5%        5559      2.504084   2.00
whisper                     0 2       5.47   0.4%   1.8%        9449     10.941298   1.00

Scores
------
Failure rate:       0.00% (PASS)
Score:               7.61
```

**Comment by Delaunay** [comment]
It fails because `Device 0 loads is below threshold  0.33 <  0.50 (max load:  1.00)`

GPU load below threshold is fatal on the CI.

**Comment by Delaunay** [comment]
Yeah, it looks like `milabench prepare` skipped opt for some reason and the prepare gets done in the run which should not happen

**Comment by satyaog** [comment]
Ah thanks I'll look into it

**Comment by Delaunay** [comment]
Merged as part of https://github.com/mila-iqia/milabench/pull/157

**Pull request closed without merging**